<launch>
    <arg name="gui" default="False"/>

    <node pkg="giskardpy" type="giskard_trees.py" name="giskard" output="screen">
        <param name="path_to_data_folder" value="$(find giskardpy)/data"/>
        <param name="enable_gui" value="$(arg gui)"/>
        <rosparam command="load" file="$(find giskardpy)/config/donbot.yaml"/>
    </node>

    <arg name="name_space" default="whole_body_controller/base"/>
    <rosparam command="load" file="$(find omni_pose_follower)/config/ros_control_sim.yaml" />
    <node pkg="omni_pose_follower" type="omni_pose_follower.py" name="omni_pose_follower" output="screen">
        <remap from="~cmd_vel" to="/omnidrive/cmd_vel_twist"/>
        <remap from="~follow_joint_trajectory" to="/whole_body_controller/base/follow_joint_trajectory"/>
        <remap from="~state" to="/whole_body_controller/base/state"/>
        <param name="name_space" value="$(arg name_space)"/>
        <param name="vel_tolerance" value="0.01"/>
        <param name="sim" value="False"/>
    </node>

  <node pkg="giskardpy" type="joint_trajectory_splitter.py" name="joint_trajectory_splitter" output="screen">
    <rosparam param="state_topics">
      - /whole_body_controller/base/state
      - /fake_state
      - /refills_finger/state
    </rosparam>
    <rosparam param="client_topics">
      - /whole_body_controller/base/follow_joint_trajectory
      - /whole_body_controller/body/follow_joint_trajectory
      - /whole_body_controller/refills_finger/follow_joint_trajectory
    </rosparam>
    <remap from="/whole_body_controller/follow_joint_trajectory" to="/whole_body_controller/delayed_follow_joint_trajectory"/>
  </node>

  <node pkg="giskard_donbot_finger_hack" type="giskard_donbot_finger_hack.py" name="giskard_donbot_finger_hack" output="screen">
    <remap from="/body/joint_states" to="/arm/joint_states"/>
  </node>

    <node pkg="giskardpy" type="interactive_marker.py" name="giskard_interactive_marker" output="screen">
        <rosparam param="enable_self_collision">False</rosparam>
        <rosparam param="interactive_marker_chains">
            - [odom, gripper_tool_frame]
            - [ur5_shoulder_link, camera_link]
            - [odom, base_footprint]
        </rosparam>
    </node>

</launch>
